x/emissions/keeper/keeper.go

```diff
@@ -1134,21 +1128,39 @@ func (k *Keeper) AppendInference(
 		return types.ErrCantUpdateEmaMoreThanOncePerWindow
 	}
 
+	// Get lowest inferer score ema for the topic
 	lowestEmaScore, found, err := k.GetLowestInfererScoreEma(ctx, topic.Id)
 	if err != nil {
 		return errorsmod.Wrap(err, "error getting lowest inferer score ema")
-		// If there are no lowest inferer score ema, calculate it
-	} else if !found {
-		lowestEmaScore, err = GetLowestScoreFromAllInferers(ctx, k, topic.Id, workerAddresses)
-		if err != nil {
-			return errorsmod.Wrap(err, "error getting lowest score from all inferers")
-		}
+	}
+	// If there are no lowest inferer score ema, it means it is the first inference for the topic
+	if !found {
+		lowestEmaScore = previousEmaScore
 		err = k.SetLowestInfererScoreEma(ctx, topic.Id, lowestEmaScore)
 		if err != nil {
 			return errorsmod.Wrap(err, "error setting lowest inferer score ema")
 		}
 	}
```

Comment:
I think this can be removed; this condition is already checked in the code below. 

It will avoid double persistence.

Reply:
Also please double-check if we still need `UpdateLowestScoreFromInfererAddresses` in this new implementation, I can see some cases that we persist the same lowest score 3 times.

Reply:
We still do need `UpdateLowestScoreFromInfererAddresses` I believe because if we remove an actor, it may be any actor => we don't know the new lowest actor

Reply:
> this "not found" check applies for the very first time the topic begins an epoch

the condition below is already covering this case: `if uint64(len(workerAddresses)) == 0 `

Reply:
Indeed, not found is not necessary, if it's just used for the very first time.